legate/core/runtime.py

```diff
@@ -880,7 +1286,273 @@ def dispatch(self, op, redop=None):
+        fused_id = self._contexts["cunumeric"].fused_id
+        numpy_context = self._contexts["cunumeric"]
+        numpy_runtime = numpy_context._library.runtime
+        z=0
+        new_op_list = []
+        for i,fusable_set in enumerate(fusable_sets):
+            start, end = fusable_set
+            op_subset = ops[start:end]
+            #if nothing to fuse, just use the original op
+            #self._fusedOpLens.append(len(op_subset))
+            if end-start==1:
+                normal_op = ops[start]
+                normal_op.strategy =  partitions[z]._strategy#uper_strategies[i]
+                new_op_list.append(normal_op)
+                z+=1
+            elif end-start > 1:
+                #initialize fused task
+                fused_task = numpy_context.create_task(fused_id)
+                #serialize necessary metadata on all encapsulated ops 
+                #this metadata will be fed into the fused op as inputs
+                meta_maps, fusion_metadata = self.serialize_multiop_metadata(numpy_context, op_subset)
+                fused_task.add_fusion_metadata(fusion_metadata) #sets fused_task._is_fused to true
+                #add typical inputs and outputs of all subtasks to fused task
+                key_part = None
+                fused_task._unfused_input_parts = []
+                fused_task._unfused_output_parts = []
+                fused_task._unfused_reduction_parts = []
+                #isScalarConversion = len(op._outputs) ==1 and len(op._inputs)==1
+                for j,op in enumerate(op_subset):
+                    for scalar in op._scalar_args:
+                        fused_task.add_scalar_arg(scalar[0], ty.int32)
+                    for (reduction, redop), part in zip(op._reductions, op._reduction_parts):
+                        fused_task.add_reduction(reduction, redop)
+                        fused_task._unfused_reduction_parts.append(part)
+                    for input,part in zip(op._inputs, op._input_parts):
+                        #print(j, part)
+                        fused_task.add_input(input)   
+                        fused_task._unfused_input_parts.append(part)
+                    for output,part in zip(op._outputs, op._output_parts):
+                        fused_task.add_output(output)   
+                        fused_task._unfused_output_parts.append(part)
+                    for future in op._futures:
+                        fused_task.add_future(future)
+                    z+=1
+                new_op_list.append(fused_task)
+        strats=[]
+        """
+        stoptime = datetime.datetime.now()
+        delta=stoptime-starttime
+        total = delta.total_seconds() * 1000.0
+        print("buildtime", total)
+        """
+   
+        #pr.enable()
+        redoPar=False
+        #starttime = datetime.datetime.now()
+        for i,fused_task in enumerate(new_op_list):
+            if redoPar:
```

Comment:
we currently have the option of redo-ing the partition for the fused op. But I will remove this